server/enhancedDate.js

```diff
@@ -1 +1,47 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+var dayNames = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+var monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 
+                        'September', 'October', 'November', 'December'];
+
+var enhancedDate = {
+    setDate: function setDate(date) {
+        if (date instanceof Date || typeof date === 'number') {
+            myDate = new Date(date);    // create a new date object so we can't overwrite the myDate var declared above
+        } else {
+            myDate = new Date();
+        }
+    },
+
+    getDate: function getDate(returnObj) {
+        if (returnObj) {
```

Comment:
All of the non-setDate methods should check if myDate is set first before trying to use it.